### PR TITLE
Add ref releases

### PR DESCRIPTION
**Pull request by MarcelMue**
## Checklist

- [ ] Consider SIG UX feedback.
- [ ] Update changelog in CHANGELOG.md.
- [x] If adding a new type, ensure that it is added to the [CRD unit tests](https://github.com/giantswarm/apiextensions/blob/d78aba91d578d56ef49f58e336035d35edc4e1b0/pkg/crd/crd_test.go#L9).

### CODE REVIEW

File: config/crd/v1beta1/release.giantswarm.io_releases.yaml [hunk 1]
```diff
@@ -76,9 +77,18 @@ spec:
               description: Components describes components used in this release.
               items:
                 properties:
+                  catalog:
```
Comment (kopiczko) [inline config/crd/v1beta1/release.giantswarm.io_releases.yaml:80]: Do we need this?

File: config/crd/v1beta1/release.giantswarm.io_releases.yaml [hunk 2]
```diff
@@ -76,9 +77,18 @@ spec:
...
+                    default: control-plane-catalog
+                    description: Catalog specifies the name of the app catalog that
+                      this component belongs to.
+                    type: string
                   name:
                     description: Name of the component.
                     type: string
+                  reference:
+                    description: Reference is the git tag or SHA specifying the component
```
Comment (kopiczko) [inline config/crd/v1beta1/release.giantswarm.io_releases.yaml:89]: I guess you don't mean SHA. It's always the version in catalog. In test catalog you'll have versions like `0.0.1-921feaa96a9882b974755c87eb8d852afb6e173a`.

Comment (MarcelMue) [reply config/crd/v1beta1/release.giantswarm.io_releases.yaml:89]: Yes - description is not accurate yet :D

File: config/crd/v1beta1/release.giantswarm.io_releases.yaml (hunk 1 above)
Comment (MarcelMue) [reply config/crd/v1beta1/release.giantswarm.io_releases.yaml:80]: So current thinking is we do need this - it is more clear then to define if a reference is from a different catalog. Without this we would need some implicit magic to determine the correct catalog which we already do in opsctl. So I would not like to reimplement that somewhere else.

File: config/crd/v1beta1/release.giantswarm.io_releases.yaml (hunk 2 above)
Comment (stone-z) [reply config/crd/v1beta1/release.giantswarm.io_releases.yaml]: Updated

**Comment by stone-z** [comment]
Since this is only additions, I _believe_ it is backwards-compatible

**Comment by MarcelMue** [comment]
Please fix merge conflicts, then this should be ready to go

**Comment by stone-z** [comment]
Oh boy this merge might be ugly. Here goes nothing

**Comment by stone-z** [comment]
Huh. everything-went-better-than-expected.png

**Comment by MarcelMue** [comment]
We should add something in the changelog afterwards :D

**Comment by stone-z** [comment]
Whoops. https://github.com/giantswarm/apiextensions/pull/521

**Pull request merged**